mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala

```diff
@@ -777,15 +778,34 @@ class ALSSuite
   }
 }
 
-class ALSCleanerSuite extends SparkFunSuite {
+class ALSCleanerSuite extends SparkFunSuite with BeforeAndAfterEach {
+
+  private var localDir: File = _
```

Comment:
Yea, but to be more clear, I guess that is cached to `Utils.localRootDirs` when we call `Utils.getLocalDir` first. So, `Utils.localRootDirs` has a kind of the state actually but yea it sounds best to not have a state in _test class_.